### PR TITLE
Add contentSynchronisation field to remoteRepository struct

### LINKED ISSUE #39: Add support for configuring replications
# Below are required fields for qualifying your issue to our support team

## Subject of the issue

- [ ] Add support for configuring replications

## Your environment

- [ ] Artifactory 6.15.0

## Expected behavior

- [ ] CRUD support for replications
- [ ] Either a separate replications service or attach get/set methods to local and remote repository structs for push and pull replications, respectively. 
- [ ] [Artifactory+REST+API#ArtifactoryRESTAPI-GetRepositoryReplicationConfiguration](http://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-GetRepositoryReplicationConfiguration)
- [ ] [Artifactory+REST+API#ArtifactoryRESTAPI-SetRepositoryReplicationConfiguration](https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-SetRepositoryReplicationConfiguration)
- [ ] [Artifactory+REST+API#ArtifactoryRESTAPI-UpdateRepositoryReplicationConfiguration](https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-UpdateRepositoryReplicationConfiguration)
- [ ] [Artifactory+REST+API#ArtifactoryRESTAPI-DeleteRepositoryReplicationConfiguration](https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-DeleteRepositoryReplicationConfiguration)
- [ ] [Artifactory+REST+API#ArtifactoryRESTAPI-CreateLocalMulti-pushReplication](https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-CreateLocalMulti-pushReplication)
- [ ] [Artifactory+REST+API#ArtifactoryRESTAPI-UpdateLocalMulti-pushReplication](https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-UpdateLocalMulti-pushReplication)
- [ ] [Artifactory+REST+API#ArtifactoryRESTAPI-DeleteLocalMulti-pushReplication](https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-DeleteLocalMulti-pushReplication)
- [ ] [Artifactory+REST+API#ArtifactoryRESTAPI-CreateorReplaceLocalMulti-pushReplication](https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-CreateorReplaceLocalMulti-pushReplication)
- [ ] [Replication+Configuration+JSON](https://www.jfrog.com/confluence/display/RTF/Replication+Configuration+JSON)

## Actual behavior

N/A

## Steps to Reproduce

N/A

### PR DESCRIPTION
**Pull request by joeh90**
This field is important for configuring smart remote repositories. Closes #39

### CODE REVIEW

File: artifactory/repositories.go [hunk 1]
```diff
@@ -95,44 +95,59 @@ func (l LocalRepository) String() string {
...
+	DownloadRedirect                  *bool                   `json:"downloadRedirect,omitempty"`
+	FeedContextPath                   *string                 `json:"feedContextPath,omitempty"`
+	DownloadContextPath               *string                 `json:"downloadContextPath,omitempty"`
+	V3FeedUrl                         *string                 `json:"v3FeedUrl,omitempty"`
+	XrayIndex                         *bool                   `json:"xrayIndex,omitempty"`
+	ContentSynchronisation            *contentSynchronisation `json:"contentSynchronisation,omitempty"`
+}
+
+// ContentSynchronisation represents smart remote repository configuration
+type contentSynchronisation struct {
```
Comment (JordanSussman) [inline artifactory/repositories.go]: struct should be exported

```suggestion
type ContentSynchronisation struct {
```

File: artifactory/repositories.go [hunk 2]
```diff
@@ -95,44 +95,59 @@ func (l LocalRepository) String() string {
...
+	BypassHeadRequests                *bool                   `json:"bypassHeadRequests,omitempty"`
+	ClientTLSCertificate              *string                 `json:"clientTlsCertificate,omitempty"`
+	ExternalDependenciesEnabled       *bool                   `json:"externalDependenciesEnabled,omitempty"`
+	ExternalDependenciesPatterns      *[]string               `json:"externalDependenciesPatterns,omitempty"`
+	DownloadRedirect                  *bool                   `json:"downloadRedirect,omitempty"`
+	FeedContextPath                   *string                 `json:"feedContextPath,omitempty"`
+	DownloadContextPath               *string                 `json:"downloadContextPath,omitempty"`
+	V3FeedUrl                         *string                 `json:"v3FeedUrl,omitempty"`
+	XrayIndex                         *bool                   `json:"xrayIndex,omitempty"`
+	ContentSynchronisation            *contentSynchronisation `json:"contentSynchronisation,omitempty"`
```
Comment (JordanSussman) [inline artifactory/repositories.go]:
```suggestion
	ContentSynchronisation            *ContentSynchronisation `json:"contentSynchronisation,omitempty"`
```

File: artifactory/repositories_test.go [hunk 3]
```diff
@@ -190,6 +190,18 @@ func Test_Repositories(t *testing.T) {
 					V3FeedUrl:                         String("https://api.nuget.org/v3/index.json"),
 					FeedContextPath:                   String("api/v2"),
 					XrayIndex:                         Bool(false),
+					ContentSynchronisation: &contentSynchronisation{
```
Comment (JordanSussman) [inline artifactory/repositories_test.go]:
```suggestion
					ContentSynchronisation: &ContentSynchronisation{
```

File: artifactory/repositories.go (hunk 1 above)
Comment (joeh90) [reply artifactory/repositories.go]: Happy to change. This was intentional as I didn't think it should be accessed outside of the package, i.e. it should only accessible via the ContentSynchronisation field of the remoteRepository struct.

Comment (jbrockopp) [reply artifactory/repositories.go]: > This was intentional as I didn't think it should be accessed outside of the package

I do agree that the field shouldn't really be accessed outside the package.

**However**, we currently have established a pattern where we still make it exported. Here are some samples:

* https://github.com/target/go-arty/blob/master/artifactory/artifacts.go#L42-L44
* https://github.com/target/go-arty/blob/master/artifactory/build.go#L31-L42
* https://github.com/target/go-arty/blob/master/artifactory/licenses.go#L52-L54

The simplest reason I can give for why we have this pattern is mainly due to the **potential** for a consumer of this library to want to directly access that field vs using the getter (i.e. `GetContentSynchronisation()`).

Based off that, I think it'd be great if you could make the struct exported 👍

**Comment by jbrockopp** [comment]
🎉 @joeh90 thanks for your contribution! 🎉

File: artifactory/repositories.go (hunk 1 above)
Comment (joeh90) [reply artifactory/repositories.go]: No problem at all. :)

File: artifactory/repositories.go (hunk 2 above)
Comment (joeh90) [reply artifactory/repositories.go]: Added in 1d85f67

File: artifactory/repositories_test.go (hunk 3 above)
Comment (joeh90) [reply artifactory/repositories_test.go]: Added in 1d85f67.

**Pull request merged**